projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-ai.php

```diff
@@ -117,6 +119,137 @@ public function register_basic_routes() {
 				),
 			)
 		);
+
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/jetpack-search/ai/search',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'request_chat_with_site' ),
+					'permission_callback' => '__return_true',
+				),
+				'args' => array(
+					'query'         => array(
+						'description'       => 'Your question to the site',
+						'required'          => true,
+						'sanitize_callback' => 'sanitize_text_field',
+					),
+					'answer_prompt' => array(
+						'description'       => 'Answer prompt override',
+						'required'          => false,
+						'sanitize_callback' => 'sanitize_text_field',
+					),
+				),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/jetpack-search/ai/rank',
+			array(
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'rank_response' ),
+					'permission_callback' => '__return_true',
+				),
+				'args' => array(
+					'cache_key' => array(
+						'description'       => 'Cache key of your response',
+						'required'          => true,
+						'sanitize_callback' => 'sanitize_text_field',
+					),
+					'comment'   => array(
+						'description'       => 'Optional feedback',
+						'required'          => false,
+						'sanitize_callback' => 'sanitize_text_field',
+					),
+					'rank'      => array(
+						'description'       => 'How do you rank this response',
+						'required'          => false,
+						'sanitize_callback' => 'sanitize_text_field',
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Get a response from chatting with the site.
+	 * Uses Jetpack Search.
+	 *
+	 * @param  WP_REST_Request $request The request.
+	 * @return mixed
+	 */
+	public function request_chat_with_site( $request ) {
+		$question = $request->get_param( 'query' );
+		$blog_id  = \Jetpack_Options::get_option( 'id' );
+		$response = Client::wpcom_json_api_request_as_blog(
+			sprintf( '/sites/%d/jetpack-search/ai/search', $blog_id ) . '?force=wpcom',
+			2,
+			array(
+				'method'  => 'GET',
+				'headers' => array( 'content-type' => 'application/json' ),
+				'timeout' => 30,
+			),
+			array(
+				'query'         => $question,
+				/**
+				 * Filter for an answer prompt override.
+				 * Example: "Talk like a cowboy."
+				 *
+				 * @param string $prompt_override The prompt override string.
+				 *
+				 * @since $$next-version$$
+				 */
+				'answer_prompt' => apply_filters( 'jetpack_ai_chat_answer_prompt', false ),
```

Comment:
Are we calling this "AI Chat" or "AI Search" ?
AI Search seems a little more honest, because this is one-shot AND technically built on top of Jetpack Search